### ISSUE TITLE
Error with regular expressions on index

### ISSUE THREAD

**Issue by rogerwaldvogel** [neo4j/neo4j#9434]
- Neo4j version: 3.2.0 (This example was working on 3.1.1)
- Operating system: (Windows 7)
- API/Driver: Neo4j Browser

**Execute on an empty database following commands:**

```
create (:Page {title: "Robert Waggoner - 'Lucid Dreaming - Gateway To The Inner' - Interview by Iain McNay"});
create (:Page {link:'http://www.gluecksschule.ch/gluecksschule.ch/Projekt.html'})
CREATE INDEX ON :Page(title);
CREATE INDEX ON :Page(link);
```

**Then execute the following query:**
```
match (page:Page) where page.title =~ '(?i).*rob.*' OR page.link =~ '(?i).*ro.*'
return page;
```

**This results in the following error:**

Neo.DatabaseError.Statement.ExecutionFailed
Expected 
RegularPlannerQuery(QueryGraph(Set(),Set(IdName(page)),Set(),Selections(Set(Predicate(Set(IdName(page)),Ors(Set(RegexMatch(Property(Variable(page),PropertyKeyName(title)),Parameter( AUTOSTRING0,String)), RegexMatch(Property(Variable(page),PropertyKeyName(link)),Parameter( AUTOSTRING1,String))))), Predicate(Set(IdName(page)),HasLabels(Variable(page),List(LabelName(Page)))))),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(page -> Variable(page)),QueryShuffle(List(),None,None)),None) 

Instead, got: 
RegularPlannerQuery(QueryGraph(Set(),Set(IdName(page)),Set(),Selections(Set(Predicate(Set(IdName(page)),PartialPredicateWrapper(FunctionInvocation(Namespace(List()),FunctionName(EXISTS),false,Vector(Property(Variable(page),PropertyKeyName(title)))),RegexMatch(Property(Variable(page),PropertyKeyName(title)),Parameter( AUTOSTRING0,String)))), Predicate(Set(IdName(page)),PartialPredicateWrapper(FunctionInvocation(Namespace(List()),FunctionName(EXISTS),false,Vector(Property(Variable(page),PropertyKeyName(link)))),RegexMatch(Property(Variable(page),PropertyKeyName(link)),Parameter( AUTOSTRING1,String)))), Predicate(Set(IdName(page)),HasLabels(Variable(page),List(LabelName(Page)))), Predicate(Set(IdName(page)),Ors(Set(RegexMatch(Property(Variable(page),PropertyKeyName(title)),Parameter( AUTOSTRING0,String)), RegexMatch(Property(Variable(page),PropertyKeyName(link)),Parameter( AUTOSTRING1,String))))))),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(page -> Variable(page)),QueryShuffle(List(),None,None)),None)

**Comment by systay** [comment]
This should solve the problem: https://github.com/neo4j/neo4j/pull/9438

**Comment by craigtaverner** [comment]
#9438 is merged, and should come out with 3.2.1
